modules/styling/lib/cs.ts

```diff
@@ -0,0 +1,355 @@
+  // enough that it makes sense to automatically wrap.
+  if (typeof value === 'string' && value.startsWith('--')) {
+    return `var(${value})` as any as T;
+  }
+  return value;
+}
+
+/**
+ * Walks through all the properties and values of a style and converts properties and/or values that
+ * need special processing. An example might be using a CSS variable without a `var()` wrapping.
+ */
+function convertAllProperties<T extends unknown>(obj: T): T {
+  if (typeof obj === 'object') {
+    const converted = {};
+    for (const key in obj) {
+      if ((obj as Object).hasOwnProperty(key)) {
+        (converted as any)[key] = convertAllProperties(obj[key]);
+      }
+    }
+    return converted as T;
+  }
+  return convertProperty(obj);
+}
+
+export type CS = string | Record<string, string>;
+
+export type CsVarsMap<T extends string, ID extends string | never> = [ID] extends [never]
+  ? Record<T, string>
+  : {[K in T]: `--${ID}-${K}`};
+
+export type CsVars<T extends string, ID extends string | never> = CsVarsMap<T, ID> & {
+  (input: Partial<Record<T, string>>): Record<T, string>;
+};
+
+/**
+ * Take a CSS Variable name and return a variable property
+ *
+ * ```ts
+ * const myVars = createVars('color')
+ *
+ * const myStyles = createStyles({
+ *   color: cssVar(myVars.color) // color: 'var(--{hash}-color)'
+ * })
+ * ```
+ *
+ * It can also support an optional fallback. Fallbacks should only be used if it is reasonable to
+ * expect a CSS variable isn't defined.
+ * ```ts
+ * const myStyles = createStyles({
+ *   color: cssVar(myVars.color, 'red') // color: 'var(--{hash}-color, red)'
+ * })
+ * ```
+ *
+ * If the project is set up for parsing with fallback files, a fallback will automatically be filled
+ * during the parsing phase. This is helpful for cases when CSS variables are expected, but not set
+ * in the environment.
+ */
+export function cssVar(input: string, fallback?: string) {
+  return fallback
+    ? `var(${input}, ${fallback.startsWith('--') ? `var(${fallback})` : fallback})`
+    : `var(${input})`;
+}
+
+/**
+ * Create temporary CSS variables to use in components. The CSS variable names will
+ * be unique at runtime to avoid collisions. The return value is a function and a
+ * Map. The function can be used to pass in values from JavaScript. The function will
+ * return a map of variable keys to CSS Variable names.
+ *
+ * ```ts
+ * // creates a `color` and `background` variable
+ * const myVars = createVars('color', 'background')
+ *
+ * // 'color' is a typed property. The type is `string`
+ * console.log(myValues.color) // `'var(--{hash}-color)'`
+ *
+ * // 'color' is a typed property. The type is `string?`
+ * // The returned object can be assigned to the `style` property of an element
+ * console.log(myVars({ color: 'red' })) //  `{'--{hash}-color': 'red'}`
+ *
+ * const div = document.createElement('div')
+ * div.style = myVars({ color: 'red' }) // <div style="--{hash}-color: red;" />
```

Comment:
so you've used `createVars` and then here you're using your variable and assigning it a value to use in a component right?